threadQueue.h

```diff
@@ -0,0 +1,61 @@
+//	$Header$
+
+//	********************************************************
+//	File Name:			threadQueue.h        
+//	Date:               3/19/2021
+//	Author:             Proshyn Nazarii
+//	Description:        thread-safe structure <queue> with 
+//                      the ability to simultaneously write
+//                      from multiple threads
+//	********************************************************
+
+
+#pragma once
+
+#include <mutex>
+#include <memory>
+#include <thread>
+#include <condition_variable>
+#include <queue>
+#include <utility>
+#include <functional>
+
+class ThreadQueue
+{
+public:
+
+	ThreadQueue() = default;
+
+	//! \brief                 Inserts a new element at the end of the queue
+	//! \param [in] value      New item for push 
+	//! \return 
+	void push(std::function<void()> value);
+
+	//! \brief                 Removes the next element in the queue after    
+	//!                        checking for items
+	//! \return                Shares pointer to the beginning of the queue
+	std::shared_ptr<std::function<void()>> Pop();
+
+	//! \brief                     Assigns first element of queue to the variable
+	//! \param [in,out] value      function performed in the thread 
+	//! \return                    returns result of deleting first element
+	bool Pop(std::function<void()>& value);
```

Comment:
Why don't you mention that pop return the first element from the queue, and it also removes the first element from the queue
And you mentioned nothing about blocking (mutext)